### ISSUE TITLE
Pytables wheels on macos/arm64 fail to load due to missing symbols

### ISSUE THREAD

**Issue by froody** [PyTables/PyTables#1114]
Pytables fails to load on macos/arm64 for at least python 3.10 and 3.11, see repro steps below. For both python3.11 and python3.10 I was able to fix this by running
`LD_LIBRARY_PATH=/opt/homebrew/lib python -m build` in the root of this repro. I wasn't able to find the root cause for the 3.10 failure. For 3.11 the failure is because by default utilsextension fails to find libhdf5 at link time and thus doesn't establish a link dependency as shown by `otool -l ...`

```
% otool -l ~/miniconda3/envs/test_env_311/lib/python3.11/site-packages/tables/utilsextension.cpython-311-darwin.so
...
Load command 9
      cmd LC_SOURCE_VERSION
  cmdsize 16
  version 0.0
Load command 10
          cmd LC_LOAD_DYLIB
      cmdsize 56
         name @rpath/libblosc2.2.dylib (offset 24)
   time stamp 2 Thu Jan  1 10:00:02 1970
      current version 2.13.1
compatibility version 2.0.0
Load command 11
          cmd LC_LOAD_DYLIB
      cmdsize 56
         name /usr/lib/libSystem.B.dylib (offset 24)
   time stamp 2 Thu Jan  1 10:00:02 1970
      current version 1336.61.1
compatibility version 1.0.0
Load command 12
          cmd LC_RPATH
      cmdsize 56
         path /Users/tbirch/miniconda3/envs/test_env_311/lib (offset 12)
Load command 13
...
```

After building it myself above I see
```
Load command 10
          cmd LC_LOAD_DYLIB
      cmdsize 72
         name /opt/homebrew/opt/hdf5/lib/libhdf5.310.dylib (offset 24)
   time stamp 2 Thu Jan  1 10:00:02 1970
      current version 310.3.0
compatibility version 310.0.0
```

## Repro steps
```
% ﻿conda create --name test_env_310 python=3.10
...
% conda activate test_env_310
% python -m pip install tables
...
% python -c "import tables"
﻿Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/tbirch/miniconda3/envs/mytestnv10/lib/python3.10/site-packages/tables/__init__.py", line 81, in <module>
    from .file import File, open_file, copy_file
  File "/Users/tbirch/miniconda3/envs/mytestnv10/lib/python3.10/site-packages/tables/file.py", line 22, in <module>
    from . import hdf5extension
ImportError: dlopen(/Users/tbirch/miniconda3/envs/mytestnv10/lib/python3.10/site-packages/tables/hdf5extension.cpython-310-darwin.so, 0x0002): symbol not found in flat namespace '_b2nd_copy_buffer'

% ﻿conda create --name test_env_311 python=3.11
...
% conda activate test_env_311
% python -m pip install tables
...
% python -c "import tables"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/tbirch/miniconda3/envs/mytestnv/lib/python3.11/site-packages/tables/__init__.py", line 44, in <module>
    from .utilsextension import get_hdf5_version as _get_hdf5_version
ImportError: dlopen(/Users/tbirch/miniconda3/envs/mytestnv/lib/python3.11/site-packages/tables/utilsextension.cpython-311-darwin.so, 0x0002): symbol not found in flat namespace '_H5E_CALLBACK_g'
```

**Comment by froody** [comment]
I think the most reliable workaround would be to include the libhdf5 binary in the wheel as this removes the need for the user to install it separately.

Otherwise somehow adding "-L/opt/homebrew/lib" to `LFLAGS` by default when building for macos/arm64, with an optional flag to disable this behavior for local builds.

Additionally, using `-undefined error` instead of `-undefined dynamic_lookup` when linking would surface this error at building time rather than post-install.

**Comment by avalentino** [comment]
Hi @froody
providing wheels that work reliably on all platforms is not easy especially for packages that depend on external C/C++ dynamic libraries.
Currently the PyTables project does not provides wheels for macos/arm64.
There is a long standing attempt to produce such wheels in PR #1050.
Any help on that side would be really appreciated.

Regarding you first point, I agree with your approach. Setting the environment to allow the compiler and the `setup.py` script to find the needed libraries is the way to go. It soul be also in our documentation if I remember correctly.

Regarding the second point, about the `-undefined` flag, I think that the `setup.py` script and the compiler already make a very good job in detecting missing libraries. By the way I'm not super expert with the osx dev environment so if you think that the flag could help, please feel free to submit a PR.

**Comment by avalentino** [comment]
This should be fixed with #1050.
